PWGHF/HFL/Tasks/HFmAnalysis.cxx

```diff
@@ -0,0 +1,340 @@
+
+constexpr static uint32_t gMuonFillMap(VarManager::ObjTypes::Muon | VarManager::ObjTypes::MuonCov);
+constexpr static uint32_t gEventFillMap(VarManager::ObjTypes::BC | VarManager::ObjTypes::Collision);
+constexpr static uint32_t gTrackMCFillMap(VarManager::ObjTypes::ParticleMC);
+
+/// Defines the histograms for classes required in analysis.
+void defineHistograms(HistogramManager*, TString);
+
+struct HfmEventSelection {
+  Configurable<bool> applySoftwareTrigger{"appllySoftwareTrigger", false, "whether to apply the software trigger"};
+  Configurable<int> softwareTrigger{"softwareTrigger", VarManager::kIsMuonSingleLowPt7, "software trigger flag"};
+  Configurable<bool> applyCutZVtx{"applyCutZVtx", false, "whether to apply the VtxZ cut"};
+  Configurable<float> zVtxMin{"zVtxMin", -10., "min. z of primary vertex [cm]"};
+  Configurable<float> zVtxMax{"zVtxMax", 10., "max. z of primary vertex [cm]"};
+
+  HistogramManager* histMan;
+  OutputObj<THashList> outputList{"output"};
+
+  float* values;
+  AnalysisCompositeCut* eventCut;
+
+  void init(o2::framework::InitContext&)
+  {
+    VarManager::SetDefaultVarNames();
+    values = new float[VarManager::kNVars];
+    eventCut = new AnalysisCompositeCut("event selection", "Event Selection", true);
+
+    AnalysisCut cut;
+    if (applyCutZVtx)
+      cut.AddCut(VarManager::kVtxZ, zVtxMin, zVtxMax, false);
+    if (applySoftwareTrigger)
+      cut.AddCut(softwareTrigger, 0.5, 1.5, false);
+    eventCut->AddCut(&cut);
+
+    histMan = new HistogramManager("analysisHistos", "aa", VarManager::kNVars);
+    histMan->SetUseDefaultVariableNames(kTRUE);
+    histMan->SetDefaultVarNames(VarManager::fgVariableNames, VarManager::fgVariableUnits);
+
+    defineHistograms(histMan, "EventBeforeCuts;EventAfterCuts;");
+    VarManager::SetUseVars(histMan->GetUsedVars());
+    outputList.setObject(histMan->GetMainHistogramList());
+  }
+
+  Produces<aod::EventCuts> eventSel;
+
+  template <uint32_t TEventFillMap, typename TEvent>
+  void runEventSel(TEvent const& event, aod::BCs const& bcs)
+  {
+    // select events
+    VarManager::ResetValues(0, VarManager::kNEventWiseVariables);
+    VarManager::FillEvent<TEventFillMap>(event, values);
+    histMan->FillHistClass("EventBeforeCuts", values);
+
+    if (eventCut->IsSelected(values)) {
+      histMan->FillHistClass("EventAfterCuts", values);
+      eventSel(1);
+    } else {
+      eventSel(0);
+    }
+  }
+
+  void processEventDataAO2D(MyCollisions::iterator const& event, aod::BCs const& bcs)
+  {
+    runEventSel<gEventFillMap>(event, bcs);
+  }
+
+  void processEventMCAO2D(MyMcCollisions::iterator const& event, aod::BCs const& bcs)
+  {
+    runEventSel<gEventFillMap>(event, bcs);
+  }
+
+  PROCESS_SWITCH(HfmEventSelection, processEventDataAO2D, "run event selection with Data AO2D file", true);
+  PROCESS_SWITCH(HfmEventSelection, processEventMCAO2D, "run event selection with MC AO2D file", false);
```

Comment:
Mentioning AO2D is unnecessary, since you cannot run on anything else.
```suggestion
  void processEvent(MyCollisions::iterator const& event, aod::BCs const& bcs)
  {
    runEventSel<gEventFillMap>(event, bcs);
  }

  void processEventMC(MyMcCollisions::iterator const& event, aod::BCs const& bcs)
  {
    runEventSel<gEventFillMap>(event, bcs);
  }

  PROCESS_SWITCH(HfmEventSelection, processEvent, "run event selection with real data", true);
  PROCESS_SWITCH(HfmEventSelection, processEventMC, "run event selection with MC data", false);
```